### ISSUE TITLE
add_worklog doesn't respect the "user" argument.

### ISSUE THREAD

**Issue by reformstudios** [pycontribs/jira#1181]
### Bug summary

As noted in the code, there's a bug where the add_worklog method only creates a worklog for the currently logged in user (eg an api user) rather than a user specified by name or id. This renders api updates of worklogs rather broken. 

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Jira Instance type

Jira Cloud (Hosted by Atlassian)

### jira-python version

main

### Python Interpreter version

2.7, 3.8

### Which operating systems have you used?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Reproduction steps

```python
# 1. Given a Jira client instance
jira: JIRA_
# 2. When I call the function with argument x
jira.add_worklog(user=some_user_id)
# 3.
A worklog is created but assigned to the api user, not the user supplied as an argument in the call.
```

### Stack trace

```python
n/a
```

### Expected behaviour

Worklogs should be given the "Author" defined in the add_worklog "user" argument.

**Comment by guodh1** [comment]
Yes, I also encountered this problem. Is there any way to deal with it

**Comment by jeffreywu1996** [comment]
same for me, any workarounds?
